README.md

```diff
@@ -1,16 +1,1261 @@
+│                                                                  │         │    ┆         │         │   │
+│                                                                  │   Dan   ╞════Set DNS══►│  Erin   │   │
+│                                                                  │         │              │         │   │
+│                                                                  └─────────┘              └─────────┘   │
+│                                                                                                         │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+## 1.3 A Note On Serialization
+
+The JSON examples below are given in [DAG-JSON](https://ipld.io/docs/codecs/known/dag-json/), but UCAN Task is actually defined as IPLD. This makes UCAN Task agnostic to encoding. DAG-JSON follows particular conventions around wrapping CIDs and binary data in tags like so:
+
+``` json
+// CID
+{"/": "Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD"}
+
+// Bytes (e.g. a signature)
+{"/": {"bytes": "s0m3Byte5"}}
+```
+
+This format help disambiguate type information in generic DAG-JSON tooling. However, your presentation need not be in this specific format, as long as it can be converted to and from this cleanly. As it is used for the signature format, DAG-CBOR is RECOMMENDED.
+
+## 1.4 Signatures
+
+All payloads described in this spec MUST be signed with a [Varsig](https://github.com/ChainAgnostic/varsig/).
+
+# 2 High-Level Concepts
+
+## 2.1 Roles
+
+Task adds two new roles to UCAN: invoker and executor. The existing UCAN delegator and delegate principals MUST persist to the invocation.
+
+| UCAN Field | Delegation                             | Task                      |
+|------------|----------------------------------------|---------------------------------|
+| `iss`      | Delegator: transfer authority (active) | Invoker: request task (active)  |
+| `aud`      | Delegate: gain authority (passive)     | Executor: perform task (active) |
+
+### 2.1.1 Invoker
+
+The invoker signals to the executor that a task associated with a UCAN SHOULD be performed.
+
+The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `iss` field of the contained UCAN.
+
+### 2.1.2 Executor
+
+The executor is directed to perform some task described in the UCAN by the invoker.
+
+The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
+
+## 2.2 Components 
+
+![](./diagrams/concepts.svg)
+
+### 2.2.1 Closure
+
+A [Closure](#3-closure) is like a deferred function application: a request to perform some action on a resource with specific inputs.
+
+### 2.2.2 Task
+
+A [Task](#4-task) extends a Closure with additional metadata that is not used to describe the meaning of the computation or effect to be run.
+
+### 2.2.3 Batch
+
+A [Batch](#5-batch) is a way of requesting more than one action at once.
```

Comment:
What are the implied transactional guarantees here ? Can one task fail and other succeed or does failure of one implies failing a batch ?

If later what about tasks that have effects e.g. if first task prints a paper there is no way we could undo it if the second one fails.

Reply:
At least in IPVM, the transactional semantics are all-or-nothing unless you introduce mutation into the middle of a pipeline i.e. the invocation is a transaction. That detail probably should be kept open in _this_ spec, but I could be wrong.